os/x11/native_dialogs.cpp

```diff
@@ -139,8 +140,8 @@ class FileDialogX11 : public CommonFileDialog {
         // between the given "parent" argument in this show() function
         // and the created window by the utility.
 
-        // Here we run the command and get a handle to read its
-        // stdout.
+        XFlush(X11::instance()->display());
+        // Here we run the command and get a handle to read its stdout.
```

Comment:
Thanks for your patch @Joefish, I'd recommend you to avoid touching the comment that is unrelated to this change and add a comment why XFlush() is needed:
```suggestion
        // ...Probably some comment why this is needed?...
        XFlush(X11::instance()->display());

        // Here we run the command and get a handle to read its
        // stdout.
```

Reply:
I restored the formatting of the comment and added a description for the XFlush line.

To be honest, when I tried opening a file with the native file dialog in aseprite, the mouse input was ignored/blocked and the cursor was still the same as when hovering over the 'open file' label in aseprite. So I concluded there must be pending events that haven't been sent to the X server yet before popen blocked further execution. And it worked, but is still just my best guess that this is the specific reason.